### ISSUE TITLE
Some problems with load/eject scripts ...

### ISSUE THREAD

**Issue by ecsfang** [petli/codplayer#39]
Hi Peter!

First, thanks for the application!
Saves my day - now my wife can play all the CD's without to much hazzle ... ;)

I only had one issue while setting up the player, and that was that I had to insert a delay into the scripts, loading and ejecting the CD. I'm running the software on a Ubuntu machine and had to update the scripts as follows (the script would execute, but apparently the command would be "terminated" before is was executed ...).

```
  #!/bin/bash
  /usr/local/bin/codctl -t 10 disc >/dev/null 2>&1 &
  sleep 2 # Added delay to make command execute ...
```

I also noticed that the ripping progress is shown in the player UI, but after that, the disc is continued to be read/examined for a while. It feels like a two step process, where only the first step is indicated with a progress. I.e. after 100%, the disc is still occupied for some minutes before it is released, maybe I could add a progress indication for this step as well.

Lastly, do you plan to update codplayer in near future? Thinking e.g. on the "multi-disc-linking" ...

Best regards,
Thomas

**Comment by petli** [comment]
Thanks, good to hear that this is useful!

I had issues with updating to run on a systemd-based system, since the udev subsystem had changed completely.  I think I ended up just removing the & to not run the program in the background, but there was also some fiddling with the udev rules.  I haven't updated the scripts or instructions though - thanks for the prompt.

And it is true that it's a two-phase process, once the basic ripping is done the TOC is read separately.  I don't use the web control UI much since I built the LED display, so it is possible that the web UI isn't updated to show the TOC ripping status.  I'll check.

I do intend to keep building on this, and multi-disk linking is high up.  I've had other projects taking up time, but is getting back to codplayer now.
